### DISCUSSION TITLE
link.termination_type.name intended to be translated?

### DISCUSSION THREAD

**Discussion by dreng** [netbox-community/netbox#15646] [Q&A]
Hi,

in our netbox-topology-views plugin we currently have an issue with other languages than english. We check if a link is a certian type before further processing. For this check we use link.termination_type.name. Unfortunately this member variable will be translated and we end in vanishing devices because we check english link type names against french link type names.

I wonder if this translation is intended or has been included by accident. For my understanding, the link type should not be translated. If one needs a translated name, there should be a "label" member or something similar ("visibleName", "caption"... you get the point).

See also https://github.com/netbox-community/netbox-topology-views/issues/464

It would be great if one of the NetBox devs could chime in here. I'm not sure if this is worth filing a bug.

**Comment by candlerb** [comment]
[nbshell](https://docs.netbox.dev/en/stable/administration/netbox-shell/) is your friend.

```
# /opt/netbox/venv/bin/python /opt/netbox/netbox/manage.py nbshell
...
>>> c = Cable.objects.first()
>>> c
<Cable: #19>
>>> ct = c.terminations.first()
>>> ct
<CableTermination: Cable #19 to F16>
>>> help(CableTermination)
...
>>> tt = ct.termination_type
>>> tt
<ContentType: dcim | front port>
>>> str(tt)
'dcim | front port'
>>> tt.name
'front port'
>>> tt.id
75
>>> help(ContentType)
...
>>> tt.app_label
'dcim'
>>> tt.model
'frontport'
>>> tt.model_class()
<class 'dcim.models.device_components.FrontPort'>
>>> tt.model_class().__name__
'FrontPort'
>>> tt.app_labeled_name
'dcim | front port'
```

`tt` is an instance of a ContentType model.  The "name" is a reasonable thing to translate, IMO: you might want to display one of these for human consumption.

But for a stable identifier to check look up in the config, then I think `tt.model` or `tt.app_label+"."+tt.model` would be appropriate. I would be surprised if those would be translated, but I haven't tested it.

**Reply by bitcollector1** [reply]
nb shell with Jupyter is pretty awesome too! so you can reflect back!

**Comment by dreng** [comment]
Hi Brian,

thanks for your reaction.

> [nbshell](https://docs.netbox.dev/en/stable/administration/netbox-shell/) is your friend.

Not until today ;-). Normally I use the Python Shell and the variables pane in VScode within a debugging session. This lets me have a decent visual overview about the classes' member variables and their current values and it allows me to create class instances on the fly. But I see that nbshell can be a useful alternative, especially without the need for a debugging session.

> But for a stable identifier to check look up in the config, then I think `tt.model` or `tt.app_label+"."+tt.model` would be appropriate

This is what I thought of myself and what I posted in the original issue. So, a technical solution is not a problem. The main question was if the behavior was intended. Thanks for your opinion about that.
